sympy/integrals/prde.py

```diff
@@ -441,6 +441,42 @@ def prde_no_cancel_b_small(b, Q, n, DE):
     return f + H, A.col_join(B).col_join(C)
 
 
+def prde_cancel_liouvillian(b, Q, n, DE):
+    """
+    Pg, 237.
+    Examples of this are:
+    log(x + 1)*log(x), log(x + 1)/log(x)
+
+    """
+    m = len(Q)
+    f, A = [None]*(n + 1), [None]*(n + 1)
+
+    # Why use DecrementLevel? Below line answers that:
+    # Assuming that we can solve such problems over 'k' (not k[t])
+    with DecrementLevel(DE):
+        DE0 = DE
```

Comment:
I was trying to do what you suggested, but I don't think it would be wise to make such a change, comment by Kalevi very much sums that up I think https://gitter.im/jksuom/sympy?at=5940c478142826e972b28a06 ,

- DifferentialExtension object involves `Poly`, `list`, `GeneratorType`, etc,..., the first three of them require different methods to create a deep-copy of them, `copy.deepcopy(list)` works, but for `GeneratorType`, `Poly` it wouldn't work. We can create a deep-copy but the code would look messy and hackish.

> it would let you mix the original DE with the decremented one (which you apparently want to do here).

We can do without that for now.